### ISSUE TITLE
Add optional schema registration to `Avromatic::Model::MessagingSerialization` Module

### ISSUE THREAD

**Issue by dstrates** [salsify/avromatic#163]
## Feature Request

Update the [Avromatic::Model::MessagingSerialization](https://github.com/salsify/avromatic/blob/master/lib/avromatic/model/messaging_serialization.rb#L59-L71) module to make schema registration with the schema registry optional.

### Current Behaviour

Currently, the `MessagingSerialization` module automatically registers the Avro schemas (both key and value schemas) with the schema registry when the `register_schemas!` method is called. This behaviour is always on, with no option to disable it.

### Proposed Change

Introduce a class-level configuration option to control whether schemas should be registered automatically. This will allow developers to opt-in or opt-out of schema registration based on their specific use cases. This could potentially take the form of a class-level variable and methods to toggle the registration behaviour.

**Comment by jturkel** [comment]
Thanks for the feature request @dstrates. Can you elaborate on the use cases where you wouldn't want automatic registration of the schema? Ultimately Avromatic needs some way of resolving the schema to an integer id that can be included in the serialized message.

**Comment by dstrates** [comment]
The main use case is that some environments rely on centralised schema management, rather than letting producers handle it directly. Optional schema registration would be really helpful to control schema evolution in these environments.

I'm not very familiar with Avromatic but  I've made a commit [here](https://github.com/salsify/avromatic/compare/master...dstrates:avromatic:feat/optional-schema-registration) with some quick thoughts. I think we can fetch an existing ID associated with a schema from the registry. Appreciate any input @jturkel

**Comment by jturkel** [comment]
Makes sense. I'd be open to a PR that implements the changes if you're willing to make changes to [avro_turf](https://github.com/dasch/avro_turf) to expose the necessary schema registry lookup method in [`AvroTurf::ConfluentSchemaRegistry`](https://github.com/dasch/avro_turf/blob/master/lib/avro_turf/confluent_schema_registry.rb) (and [`AvroTurf::CachedConfluentSchemaRegistry`](https://github.com/dasch/avro_turf/blob/master/lib/avro_turf/cached_confluent_schema_registry.rb)).

**Comment by jturkel** [comment]
Also note you can configure `Avromatic.schema_registry` with an object that overrides the `register` method to do whatever you want if you're looking for a quick workaround.
